File: DIP/DIP_008/C++/DIP_008_Canny_edge_detector.cpp
#include <opencv2/opencv.hpp>
#include <iostream>

using namespace cv;
using namespace std;

int main()
{
    // Loading the actual image
    Mat image, gray, edges;

    image=imread("daenerys.jpg", IMREAD_COLOR);

    // Edge cases
    if(image.empty()){
    	cout << "Error loading image" << endl;
    	return -1;
    }

    // converting the color image into grayscale
    cvtColor(image, gray, COLOR_BGR2GRAY);

    // Detect edges with Canny Edge Detector
    // First argument is the input image.
    // Second argument is the output image.
    // Third and fourth arguments are the minVal and maxVal.
    // Fifth argument is the size of Sobel kernel used for find image gradients (aperture_size).
    // Last argument is L2gradient.
    Canny(image, edges, 100, 200, 3);

    // Displaying the result
    cv::imshow("Original image", image);
    cv::imshow("Canny Edge Detector", edges);

    cv::waitKey(0);
    return 0;
}

File: DIP/DIP_008/Python/DIP_008_Canny_edge_detector.py
import cv2

# read the image as a grayscale
image = cv2.imread('daenerys.jpg',0)

# Detect edges with Canny Edge Detector
# First argument is the input image.
# Second and third arguments are the minVal and maxVal.
# Third argument is the size of Sobel kernel used for find image gradients (aperture_size).
# Last argument is L2gradient.
edges = cv2.Canny(image,100,200,3)

cv2.imshow("Original image", image)
cv2.imshow("Canny Edge Detector", edges)
cv2.waitKey(0)
cv2.destroyAllWindows();